eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+package com.eternalcode.core.feature.jail;
+
+import com.eternalcode.commons.bukkit.position.Position;
+import com.eternalcode.commons.bukkit.position.PositionAdapter;
+import com.eternalcode.commons.time.DurationParser;
+import com.eternalcode.commons.time.TemporalAmountParser;
+import com.eternalcode.core.feature.jail.event.JailDetainEvent;
+import com.eternalcode.core.feature.jail.event.JailReleaseEvent;
+import com.eternalcode.core.feature.spawn.SpawnService;
+import com.eternalcode.core.feature.teleport.TeleportService;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Service;
+
+import com.eternalcode.core.notice.NoticeService;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class JailServiceImpl implements JailService {
+
+    private final TeleportService teleportService;
+    private final SpawnService spawnService;
+    private final NoticeService noticeService;
+    private final JailSettings settings;
+    private Position jailPosition;
+    private final JailLocationRepository jailRepository;
+
+    private final PrisonersRepository prisonersRepository;
```

Comment:
jailPosition I guess it's in the configuration. Consider whether you need to store the item
```suggestion
    private final TeleportService teleportService;
    private final SpawnService spawnService;
    private final NoticeService noticeService;
    private final JailSettings settings;
    
    private final JailLocationRepository jailRepository;
    private final PrisonersRepository prisonersRepository;
    
    private Position jailPosition;
```

Reply:
suggested change is pure cosmetic.

jailPosition is stored in database, due to setting it via command.

Reply:
> suggested change is pure cosmetic.
> 
> jailPosition is stored in database, due to setting it via command.

Better appearance -> Better code readability

Reply:
yessir, won't argue with that!

Reply:
> suggested change is pure cosmetic.
> 
> jailPosition is stored in database, due to setting it via command.

spawn is also setting via command, but stored in configuration